Compile warnings

Hello,

I'm developing a project using Visual Studio Code with PlatformIO add-on and every time I compile my code, I get these warnings. 
The hardware I use is Arduino Nano (ATMega328p).

```
In file included from .pio\libdeps\NanoaATmega328\LcdMenu\src/LcdMenu.h:33:0,
                 from src\main.cpp:6:
.pio\libdeps\NanoaATmega328\LcdMenu\src/MenuItem.h: In constructor 'MenuItem::MenuItem(char*, char*, fptr, byte)':
.pio\libdeps\NanoaATmega328\LcdMenu\src/MenuItem.h:86:20: warning: 'MenuItem::value' will be initialized after [-Wreorder]
     String value = "";
                    ^
In file included from c:\users\kulverstukas\.platformio\packages\toolchain-atmelavr\avr\include\avr\eeprom.h:50:0,
                 from C:\users\kulverstukas\.platformio\packages\framework-arduino-avr\cores\arduino/USBAPI.h:25,
                 from C:\users\kulverstukas\.platformio\packages\framework-arduino-avr\cores\arduino/Arduino.h:233,
                 from src\main.cpp:1:
.pio\libdeps\NanoaATmega328\LcdMenu\src/MenuItem.h:48:21: warning:   'void (* MenuItem::callback)()' [-Wreorder]
     fptr callback = NULL;
                     ^
In file included from .pio\libdeps\NanoaATmega328\LcdMenu\src/LcdMenu.h:33:0,
                 from src\main.cpp:6:
.pio\libdeps\NanoaATmega328\LcdMenu\src/MenuItem.h:65:5: warning:   when initialized here [-Wreorder]
     MenuItem(char* text, char* value, fptr callback, byte type)
     ^
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+.pio
 .vscode/
-src/app.ino
+src/main.cpp
 debug.log

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=LcdMenu
-version=1.1.0
+version=1.1.1
 author=Forntoh Thomas <thomasforntoh@gmail.com>
 maintainer=Forntoh Thomas <thomasforntoh@gmail.com>
 sentence=Display navigable menu items on your LCD display with Arduino.

--- a/platformio.ini
+++ b/platformio.ini
@@ -1,0 +1,17 @@
+; PlatformIO Project Configuration File
+;
+;   Build options: build flags, source filter
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[env:uno]
+platform = atmelavr
+board = uno
+framework = arduino
+lib_deps = 
+	chris--a/Keypad@^3.1.1
+	marcoschwartz/LiquidCrystal_I2C@^1.1.4

--- a/src/LcdMenu.h
+++ b/src/LcdMenu.h
@@ -238,8 +238,7 @@ class LcdMenu {
      * @relatesalso MenuItem
      */
     void placeCursorAtEnd(MenuItem* item) {
-        uint8_t col =
-            ((String)item->getText()).length() + 2 + item->value.length();
+        uint8_t col = strlen(item->getText()) + 2 + strlen(item->value);
         lcd->setCursor(constrain(col, 0, maxCols - 1), cursorPosition - top);
     }
 

--- a/src/LcdMenu.h
+++ b/src/LcdMenu.h
@@ -55,8 +55,8 @@ class LcdMenu {
     /**
      * Last visible item's position in the menu array
      */
-    uint8_t bottom;
-    uint8_t previousBottom;
+    uint8_t bottom = 0;
+    uint8_t previousBottom = 0;
     /**
      * Rows on the LCD Display
      */
@@ -142,14 +142,14 @@ class LcdMenu {
                     // append textOn or textOff depending on the state
                     //
                     lcd->print(":");
-                    lcd->print(item->isOn ? item->textOn : item->textOff);
+                    lcd->print(item->isOn ? item->getTextOn() : item->getTextOff());
                     break;
                 case MENU_ITEM_INPUT:
                     //
                     // append the value the value of the input
                     //
                     lcd->print(":");
-                    lcd->print(item->value.substring(
+                    lcd->print(item->getValue().substring(
                         0, maxCols - ((String)item->getText()).length() - 2));
                     break;
                 default:
@@ -238,7 +238,7 @@ class LcdMenu {
      * @relatesalso MenuItem
      */
     void placeCursorAtEnd(MenuItem* item) {
-        uint8_t col = item->getText().length() + 2 + item->value.length();
+        uint8_t col = item->getText().length() + 2 + item->getValue().length();
         lcd->setCursor(constrain(col, 0, maxCols - 1), cursorPosition - top);
     }
 
@@ -486,7 +486,7 @@ class LcdMenu {
             //
             // set the value
             //
-            item->value = text;
+            item->getValue() = text;
             //
             // repaint menu
             //

--- a/src/LcdMenu.h
+++ b/src/LcdMenu.h
@@ -238,7 +238,7 @@ class LcdMenu {
      * @relatesalso MenuItem
      */
     void placeCursorAtEnd(MenuItem* item) {
-        uint8_t col = strlen(item->getText()) + 2 + strlen(item->value);
+        uint8_t col = item->getText().length() + 2 + item->value.length();
         lcd->setCursor(constrain(col, 0, maxCols - 1), cursorPosition - top);
     }
 
@@ -486,7 +486,7 @@ class LcdMenu {
             //
             // set the value
             //
-            item->value = text.c_str();
+            item->value = text;
             //
             // repaint menu
             //
@@ -509,12 +509,12 @@ class LcdMenu {
      * @param message message to display
      * @param duration how long to display the message
      */
-    void displayNotification(char* message, unsigned int duration) {
+    void displayNotification(String message, unsigned int duration) {
         /**
          * Calculate the position to start writing
          * (centralize text)
          */
-        uint8_t centerPos = maxCols / 2 - (strlen(message) / 2);
+        uint8_t centerPos = maxCols / 2 - (message.length() / 2);
         /**
          * Set cursor potion and clear lane
          */
@@ -525,8 +525,8 @@ class LcdMenu {
          * Draw each independent character
          */
         lcd->write(0xA5);
-        for (unsigned int i = 0; i < strlen(message); i++) {
-            char character = message[i];
+        for (unsigned int i = 0; i < message.length(); i++) {
+            char character = message.charAt(i);
             lcd->write(character);
         }
         lcd->write(0xA5);

--- a/src/LcdMenu.h
+++ b/src/LcdMenu.h
@@ -487,7 +487,7 @@ class LcdMenu {
             //
             // set the value
             //
-            item->value = text;
+            item->value = text.c_str();
             //
             // repaint menu
             //

--- a/src/MenuItem.h
+++ b/src/MenuItem.h
@@ -83,7 +83,7 @@ class MenuItem {
     boolean isOn = false;
     char* textOn = (char*)"ON";
     char* textOff = (char*)"OFF";
-    String value = "";
+    char* value = "";
 };
 
 class ItemHeader : public MenuItem {

--- a/src/MenuItem.h
+++ b/src/MenuItem.h
@@ -44,46 +44,46 @@ const byte MENU_ITEM_END_OF_MENU = 7;
 //
 class MenuItem {
    private:
-    char* text = (char*)"";
+    String text = (String) "";
     fptr callback = NULL;
     MenuItem* subMenu = NULL;
     byte type = MENU_ITEM_NONE;
 
    public:
     //
+    //
+    //
+    boolean isOn = false;
+    String textOn = (String) "ON";
+    String textOff = (String) "OFF";
+    String value;
+    //
     // constructors
     //
     MenuItem() = default;
-    MenuItem(char* text) : text(text) {}
-    MenuItem(char* text, fptr callback, byte type)
+    MenuItem(String text) : text(text) {}
+    MenuItem(String text, fptr callback, byte type)
         : text(text), callback(callback), type(type) {}
-    MenuItem(char* text, fptr callback, MenuItem* subMenu, byte type)
+    MenuItem(String text, fptr callback, MenuItem* subMenu, byte type)
         : text(text), callback(callback), subMenu(subMenu), type(type) {}
     MenuItem(MenuItem* subMenu, byte type) : subMenu(subMenu), type(type) {}
-    MenuItem(char* text, MenuItem* subMenu, byte type)
+    MenuItem(String text, MenuItem* subMenu, byte type)
         : text(text), subMenu(subMenu), type(type) {}
-    MenuItem(char* text, char* value, fptr callback, byte type)
+    MenuItem(String text, String value, fptr callback, byte type)
         : text(text), value(value), callback(callback), type(type) {}
     //
     // getters
     //
-    char* getText() { return text; }
+    String getText() { return text; }
     fptr getCallback() { return callback; }
     MenuItem* getSubMenu() { return subMenu; }
     byte getType() { return type; }
     //
     // setters
     //
-    void setText(char* text) { this->text = text; }
+    void setText(String text) { this->text = text; }
     void setCallBack(fptr callback) { this->callback = callback; }
     void setSubMenu(MenuItem* subMenu) { this->subMenu = subMenu; }
-    //
-    //
-    //
-    boolean isOn = false;
-    char* textOn = (char*)"ON";
-    char* textOff = (char*)"OFF";
-    char* value = "";
 };
 
 class ItemHeader : public MenuItem {
@@ -104,27 +104,27 @@ class ItemFooter : public MenuItem {
 
 class ItemInput : public MenuItem {
    public:
-    ItemInput(char* text, char* value, fptr callback)
+    ItemInput(String text, String value, fptr callback)
         : MenuItem(text, value, callback, MENU_ITEM_INPUT) {}
 };
 
 class ItemSubMenu : public MenuItem {
    public:
-    ItemSubMenu(char* text, MenuItem* parent)
+    ItemSubMenu(String text, MenuItem* parent)
         : MenuItem(text, parent, MENU_ITEM_SUB_MENU) {}
 };
 
 class ItemToggle : public MenuItem {
    public:
-    ItemToggle(char* key, fptr callback)
+    ItemToggle(String key, fptr callback)
         : MenuItem(key, callback, MENU_ITEM_TOGGLE) {}
-    ItemToggle(char* key, char* textOn, char* textOff, fptr callback)
+    ItemToggle(String key, String textOn, String textOff, fptr callback)
         : MenuItem(key, callback, MENU_ITEM_TOGGLE) {}
 };
 
 class ItemCommand : public MenuItem {
    public:
-    ItemCommand(char* key, fptr callback)
+    ItemCommand(String key, fptr callback)
         : MenuItem(key, callback, MENU_ITEM_COMMAND) {}
 };
 

--- a/src/MenuItem.h
+++ b/src/MenuItem.h
@@ -44,19 +44,18 @@ const byte MENU_ITEM_END_OF_MENU = 7;
 //
 class MenuItem {
    private:
-    String text = (String) "";
+    String text;
+    String value;
+    String textOn;
+    String textOff;
     fptr callback = NULL;
     MenuItem* subMenu = NULL;
     byte type = MENU_ITEM_NONE;
-
    public:
     //
     //
     //
     boolean isOn = false;
-    String textOn = (String) "ON";
-    String textOff = (String) "OFF";
-    String value;
     //
     // constructors
     //
@@ -71,6 +70,8 @@ class MenuItem {
         : text(text), subMenu(subMenu), type(type) {}
     MenuItem(String text, String value, fptr callback, byte type)
         : text(text), value(value), callback(callback), type(type) {}
+    MenuItem(String text, String textOn, String textOff, fptr callback, byte type)
+        : text(text), textOn(textOn), textOff(textOff), callback(callback), type(type) {}
     //
     // getters
     //
@@ -78,6 +79,9 @@ class MenuItem {
     fptr getCallback() { return callback; }
     MenuItem* getSubMenu() { return subMenu; }
     byte getType() { return type; }
+    String getValue() { return value; }
+    String getTextOn() { return textOn; }
+    String getTextOff() { return textOff; }
     //
     // setters
     //
@@ -117,9 +121,9 @@ class ItemSubMenu : public MenuItem {
 class ItemToggle : public MenuItem {
    public:
     ItemToggle(String key, fptr callback)
-        : MenuItem(key, callback, MENU_ITEM_TOGGLE) {}
+        : MenuItem(key, "", "", callback, MENU_ITEM_TOGGLE) {}
     ItemToggle(String key, String textOn, String textOff, fptr callback)
-        : MenuItem(key, callback, MENU_ITEM_TOGGLE) {}
+        : MenuItem(key, textOn, textOff, callback, MENU_ITEM_TOGGLE) {}
 };
 
 class ItemCommand : public MenuItem {

--- a/src/MenuItem.h
+++ b/src/MenuItem.h
@@ -51,6 +51,7 @@ class MenuItem {
     fptr callback = NULL;
     MenuItem* subMenu = NULL;
     byte type = MENU_ITEM_NONE;
+
    public:
     //
     //
@@ -70,8 +71,13 @@ class MenuItem {
         : text(text), subMenu(subMenu), type(type) {}
     MenuItem(String text, String value, fptr callback, byte type)
         : text(text), value(value), callback(callback), type(type) {}
-    MenuItem(String text, String textOn, String textOff, fptr callback, byte type)
-        : text(text), textOn(textOn), textOff(textOff), callback(callback), type(type) {}
+    MenuItem(String text, String textOn, String textOff, fptr callback,
+             byte type)
+        : text(text),
+          textOn(textOn),
+          textOff(textOff),
+          callback(callback),
+          type(type) {}
     //
     // getters
     //

